harness/determined/cli/workspace.py

```diff
@@ -221,8 +230,11 @@ def create_workspace(args: argparse.Namespace) -> None:
     agent_user_group = _parse_agent_user_group_args(args)
     checkpoint_storage = _parse_checkpoint_storage_args(args)
 
-    if args.cluster_name and not args.namespace:
-        raise api.errors.BadRequestException("must provide --namespace NAMESPACE")
+    if args.cluster_name and not (args.namespace or args.auto_create_namespace):
+        raise api.errors.BadRequestException(
+            "must provide --namespace NAMESPACE or \
+            --auto-creeate-namespace"
+        )
```

Comment:
When you are up for a chat, could we talk about having a dry solution? When the logic changes, we only have to do it in one place instead of in both CLI and master.

Reply:
Sure! It definitely makes sense to leave the check in the API handler in the master; I added it to the CLI as well so that we error out sooner rather than later. But I'm open to removing it since the repeated check in master arguably makes the CLI check redundant.

Reply:
The error message from the master is not tailored for CLI, but I think as long as we have a clear helper message, we should be fine.